tests/testthat/helper-setup.R

```diff
@@ -12,6 +12,9 @@ renv_tests_setup <- function(envir = parent.frame()) {
   if (!once())
     return()
 
+  # remove automatic tasks so we can capture explicitly in tests
+  renv_task_unload()
```

Comment:
I think this is the right principle; otherwise running the tests interactively and non-interactively behaves differently. It does mean you need to remember `renv_snapshot_task()` manually, but I don't see a way around that.